File: app/front/src/components/main/dto/user/acceptUserAdminCapsuleDto.ts
import { FrameworkCapsuleDto, type FrameworkCapsuleDtoInterface } from "seijishikin-jp-normalize_common-tool";
import { PromoteAdminEntity, type PromoteAdminEntityInterface } from "../../entity/promoteAdminEntity";

interface AcceptUserAdminCapsuleDtoInterface extends FrameworkCapsuleDtoInterface {

    /** SE権限追加推薦Entity */
    promoteAdminEntity: PromoteAdminEntityInterface;

}

class AcceptUserAdminCapsuleDto extends FrameworkCapsuleDto
    implements AcceptUserAdminCapsuleDtoInterface {

    /** SE権限追加推薦Entity */
    promoteAdminEntity: PromoteAdminEntityInterface;

    constructor() {
        super();
        this.promoteAdminEntity = new PromoteAdminEntity();
    }

}

export { type AcceptUserAdminCapsuleDtoInterface, AcceptUserAdminCapsuleDto }

File: app/back/src/main/java/net/seijishikin/jp/normalize/manage/kanrensha/dto/task/AcceptUserAdminCapsuleDto.java
package net.seijishikin.jp.normalize.manage.kanrensha.dto.task;

import java.io.Serializable;

import net.seijishikin.jp.normalize.common_tool.dto.FrameworkCapsuleDto;
import net.seijishikin.jp.normalize.manage.kanrensha.entity.PromoteAdminEntity;

/**
 * SE権限追加推薦諾否Dto
 */
public class AcceptUserAdminCapsuleDto extends FrameworkCapsuleDto implements Serializable {

    /** SerialId */
    private static final long serialVersionUID = 1L;

    /** SE権限追加推薦Entity */
    private PromoteAdminEntity promoteAdminEntity;

    /**
     * SE権限追加推薦Entityを取得する
     * 
     * @return SE権限追加推薦Entity
     */
    public PromoteAdminEntity getPromoteAdminEntity() {
        return promoteAdminEntity;
    }

    /**
     * SE権限追加推薦Entityを設定する
     * 
     * @param promoteAdminEntity SE権限追加推薦Entity
     */
    public void setPromoteAdminEntity(final PromoteAdminEntity promoteAdminEntity) {
        this.promoteAdminEntity = promoteAdminEntity;
    }

}
